src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -156,6 +160,262 @@ def _fold_messages_by_uid(user, messages_traced_data):
+    we are less interested in individual messages, but in the labelled opinions, collated by labelled dataset, by
+    participant, or by project.
+
+    :param analysis_dataset_config: Analysis dataset configuration to convert.
+    :type analysis_dataset_config: src.engagement_db_to_analysis.configuration.AnalysisDatasetConfiguration
+    :return: List of all the column configurations for this analysis dataset configuration.
+    :rtype: list of core_data_modules.analysis.analysis_utils.AnalysisConfiguration
+    """
+    column_configs = []
+    for coding_config in analysis_dataset_config.coding_configs:
+        config = AnalysisConfiguration(
+            dataset_name=coding_config.analysis_dataset,
+            raw_field=analysis_dataset_config.raw_dataset,
+            coded_field=f"{coding_config.analysis_dataset}_labels",
+            code_scheme=coding_config.code_scheme
+        )
+        column_configs.append(config)
+    return column_configs
+
+
+def _analysis_dataset_config_for_message(analysis_dataset_configs, message):
+    """
+    Gets the analysis dataset configuration to use to process this message, by looking-up the configuration that refers
+    to this message's engagement db "dataset" property.
+
+    :param analysis_dataset_configs: Dataset configurations to search for the one that relates to the given message.
+    :type analysis_dataset_configs: list of src.engagement_db_to_analysis.configuration.AnalysisDatasetConfiguration
+    :param message: Message to retrieve the analysis dataset configuration for.
+    :type message: engagement_database.data_models.Message
+    :return: Analysis dataset configuration to use for this message.
+    :rtype: src.engagement_db_to_analysis.configuration.AnalysisDatasetConfiguration
+    """
+    for config in analysis_dataset_configs:
+        if message.dataset in config.engagement_db_datasets:
+            return config
+    raise ValueError(f"No analysis dataset configuration found for message '{message.message_id}', which has engagement"
+                     f"db dataset {message.dataset}")
+
+
+def _get_normalised_labels_for_code_scheme(message, code_scheme):
+    """
+    Gets the labels assigned to this message under the given `code_scheme` (or a duplicate of this code scheme),
+    with label scheme_id's normalised to the primary code scheme id in cases where the code scheme was duplicated
+    e.g. scheme_id 'scheme-abc123-1' will be re-written to 'scheme-abc123'.
+
+    :param message: Message to get the labels from.
+    :type message: engagement_database.data_models.Message
+    :param code_scheme: Code scheme to get the latest labels for.
+    :type code_scheme: core_data_modules.data_models.CodeScheme
+    :return: List of the relevant, normalised labels for the given code scheme.
+    :rtype: list of core_data_modules.data_models.Label
+    """
+    relevant_labels = []
```

Comment:
```suggestion
    message_labels = [] or normalised_labels
```
relevant is easily interpretable as normal codes only which is not the case.